### ISSUE TITLE
Celery exits immediately on startup after upgrading to 4.X

### ISSUE THREAD

**Issue by rafaelosoto** [celery/celery#5036]
## Checklist
software -> celery:4.2.1 (windowlicker) kombu:4.2.0 py:2.7.12
            billiard:3.5.0.4 py-amqp:2.3.2
platform -> system:Linux arch:64bit, ELF imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:amqp results:django_celery_beat.backends.database.DatabaseBackend

MIDDLEWARE: None
CSRF_USE_SESSIONS: False
is_overridden: <bound method Settings.is_overridden of <Settings "server.settings">>
PREPEND_WWW: False
SECURE_PROXY_SSL_HEADER: 
    ('HTTP_X_FORWARDED_PROTO', 'https')
LANGUAGE_COOKIE_AGE: None
SESSION_COOKIE_HTTPONLY: True
DEBUG_PROPAGATE_EXCEPTIONS: False
INTERNAL_IPS: []
BROKER_URL: u'amqp://guest:********@localhost:5672//'

- [ X] I have included the output of ``celery -A proj report`` in the issue.
      (if you are not able to do this, then at least specify the Celery
       version affected).
- [X ] I have verified that the issue exists against the `master` branch of Celery.

## Steps to reproduce

I cannot start the celery app at all. This app was working fine previously I am doing an update of
celery, kombu, and swapping out djcelery for django-celery-beat.

I can start the beat scheduler fine but the worker dies immediately with zero output :( I tried the FAKEFORK idea for multi which actually lets me know the worker dies but not much else.

`11945 ◯  C_FAKEFORK=1 celery multi start 1 -A batch -l debug                                                                                                                                                     ⏎
celery multi v4.2.1 (windowlicker)
> Starting nodes...
> /home/waffles/checkouts/ebis/2280f/batch/celery.py(18)<module>()
-> app.conf.update(
(Pdb) c
> /home/waffles/checkouts/ebis/2280f/batch/celery.py(18)<module>()
-> app.conf.update(
(Pdb) c
        > celery1@waffleiron: * Child terminated with exit code 1
FAILED
`

This is where the worker exits:
`virtualenvs/django/local/lib/python2.7/site-packages/celery/bin/worker.py(223)run_from_argv()->1
-> return self(*args, **options)
(Pdb) l
218             argv = sys.argv[1:] if argv is None else argv
219             # parse options before detaching so errors can be handled.
220             options, args = self.prepare_args(
221                 *self.parse_options(prog_name, argv, command))
222             self.maybe_detach([command] + argv)
223  ->         return self(*args, **options)

`

## Expected behavior

App starts up like normal or gives some indication of why the child died with C_FAKEFORK on

## Actual behavior

> celery1@waffleiron: * Child terminated with exit code 1

I've tried several combinations of celery and kombu in 4.x, this was working fine in 3.X but I needed to upgrade to support django-celery-beat. Any help is highly appreciated, or maybe some direction on how to better debug the code. I'm not sure if there's a way to pop into the child process as it is forking/starting up? maybe RDB? 

Thanks!

**Comment by rafaelosoto** [comment]
Figured it out! I had upgraded to django-celery-beat and I did a search and replace s/djcelery/django_celery_beat/g and it changed my result backend to django_celery_beat.backends.... but django_celery_beat does not offer a results backend. So the child process was attempting an importing getting an ImportError and exiting silently.

I got to the bottom of it by pdb-ing into celery/bin/worker and setting a breakpoint right before worker.start() is run and following it into there. I also had to set the C_FAKEFORK=1 environment variable.
